integration-tests/bats/checkout.bats

```diff
@@ -104,6 +104,41 @@ SQL
     [[ "$output" =~ "some error" ]] || false
 }
 
+@test "checkout: dolt checkout table to restore working tree tables with add and drop foreign key" {
+    dolt sql -q "create table t (c1 int primary key, c2 int, check(c2 > 0))"
+    dolt sql -q "create table z (c1 int primary key, c2 int)"
+    dolt commit -Am "create tables t and z"
+
+    dolt sql -q "ALTER TABLE z ADD CONSTRAINT foreign_key1 FOREIGN KEY (c1) references t(c1)"
+    run dolt status
+    [[ "$output" =~ "Changes not staged for commit:" ]] || false
+    [[ "$output" =~ "modified:       z" ]] || false
+
+    run dolt checkout z
+    [ "$status" -eq 0 ]
+    [ "$output" = "" ]
+
+    run dolt status
+    [[ "$output" =~ "On branch main" ]] || false
+    [[ "$output" =~ "nothing to commit, working tree clean" ]] || false
```

Comment:
(minor) might be worth an explicit call to `show create table` or something and grep that there is or is not the `foreign_key1` constraint name – it would just be a very direct test of the foreign key being there (or not). I'm guessing that before this change the `dolt checkout z` call either failed, or left the change in the working set, so this test makes sense, but just might be more obvious with the direct foreign key check for future readers.